### PR TITLE
Doc

### PR DESCRIPTION
**Pull request by timsaxe**
This commit sets up the documentation framework:
rtl/../data/file.csv for files that define registers
docs/Makefile copies csv into a mirror directory (there is probably a better make way, but I haven;t taken the time to find it)
docs/Makefile runs util/ioscript.py to convert the csv files into auto-doc/*.md
docs/Makefile runs Sphinx to combine doc-src/*, auto-doc/* into the document tree which can be found in _build/html

csv files and ioscript were modified to put module description before the register descriptions

overview.md was updated to include more information, especially about software validation

### CODE REVIEW

File: docs/core-v-mcu-user-manual.rst [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    :maxdepth: 1
+    
+    doc-src/overview.md
+    auto-docs/apb_soc_ctrl_reg_defs
+    auto-docs/apb_interrupt_cntrl_reg_defs
+    auto-docs/apb_timer_unit_reg_defs
+    auto-docs/apb_gpio_reg_defs
+    auto-docs/udma_ctrl_reg_defs
+    auto-docs/udma_uart_reg_defs
+    auto-docs/udma_i2cm_reg_defs
```
Comment (zarubaf) [inline docs/core-v-mcu-user-manual.rst:18]: Would you mind adding back a link to the Github page `github.rst`. We can also put it under a specific directory?

File: util/ioscript.py [hunk 2]
```diff
@@ -0,0 +1,1185 @@
+#!/bin/python3
```
Comment (zarubaf) [inline util/ioscript.py:1]: That's great. I think @gmartin102  also added the `ioscript` to another folder. Can we make sure that we only have one version in the repo (e.g., deleting the other one)? I think having it under `util/` makes the most sense.

Maybe a small license header would also be nice :-)

**Review by zarubaf — CHANGES_REQUESTED** [review]
Thanks a lot. This looks very clean!

File: docs/core-v-mcu-user-manual.rst (hunk 1 above)
Comment (timsaxe) [reply docs/core-v-mcu-user-manual.rst:18]: I was wondering what to do with github.rst.  Just need to decide where it fits.

File: util/ioscript.py (hunk 2 above)
Comment (timsaxe) [reply util/ioscript.py:1]: Licenses is more than nice if I recall Eclipse foundation requirements.  Greg and I had discussed and agreed on utils, so I need to find the other and eliminate it.

**Review by zarubaf — APPROVED** [review]

**Pull request merged**